app/Http/routes.php

```diff
@@ -1,27 +1,34 @@
 <?php
 
-Route::get('/', function () {
-    $rss = new \Vinelab\Rss\Rss;
-    $feed = $rss->feed(env('RSS_URL'));
+use App\Http\ArticleMapper;
+use Carbon\Carbon;
 
-    return view('episodes.index')
-        ->with('podcast', $feed->info())
-        ->with('episodes', $feed->articles);
-});
+Route::get( '/', function () {
+    $rss  = new \Vinelab\Rss\Rss;
+    $feed = $rss->feed( env( 'RSS_URL' ) );
 
-Route::get('{id}', function ($id) {
-    $rss = new \Vinelab\Rss\Rss;
-    $feed = $rss->feed(env('RSS_URL'));
+    $episodes = $feed->articles->map( function ( $article, $key ) {
+        return ArticleMapper::map($article);
+    } )->sortByDesc( 'sortDate' );
 
-    $episode = $feed->articles->sortBy(function ($episode, $key) {
-        return new \Carbon\Carbon($episode->pubDate);
-    })->values()->get($id - 1);
+    return view( 'episodes.index' )
+        ->with( 'podcast', $feed->info() )
+        ->with( 'episodes', $episodes );
+} );
+
+Route::get( '/{id}/{title}', function ( $id )  {
+    $rss  = new \Vinelab\Rss\Rss;
+    $feed = $rss->feed( env( 'RSS_URL' ) );
+
+    $episode = $feed->articles->map( function ( $article, $key ) {
+        return ArticleMapper::map($article);
+    } )->where('sortId', $id)->first();
```

Comment:
I don't think this works until you iterate over all of them and assign them a unique ID